test/e2e/parallel/registration_service_test.go

```diff
@@ -728,22 +730,40 @@ func TestActivationCodeVerification(t *testing.T) {
 		// check that the MUR and Space are configured as expected
 		// Wait for the UserSignup to have the desired state
 		userSignup, err = hostAwait.WaitForUserSignup(t, userSignup.Name,
-			wait.UntilUserSignupHasCompliantUsername())
+			wait.UntilUserSignupHasCompliantUsername(),
+			wait.UntilUserSignupHasTargetCluster(targetCluster))
```

Comment:
also based on the e2e test [failure here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/codeready-toolchain_toolchain-e2e/1035/pull-ci-codeready-toolchain-toolchain-e2e-master-e2e/1824264958020423680/build-log.txt):
```
  -expected
        +actual
          string(
        - 	"",
        + 	"member-ci-op-7l79p51s-ceea8.devsandboxci.devcluster.openshift2",
          )
        
        
    registration_service_test.go:735: 
        	Error Trace:	/go/src/github.com/codeready-toolchain/toolchain-e2e/test/e2e/parallel/registration_service_test.go:735
        	            				/go/src/github.com/codeready-toolchain/toolchain-e2e/test/e2e/parallel/registration_service_test.go:762
```

 seems like we need to add this condition only if the targetCluster is not empty

Reply:
It should work. I think the test will pass now that your comment https://github.com/codeready-toolchain/toolchain-e2e/pull/1035#discussion_r1719654851 was addressed. Basically because the SocialEvent was always having `testsocialevent.WithTargetCluster(member2Await.ClusterName)` then even the test with "no target cluster" actually had the TargetCluster set and it caused the UserSignup to have the target cluster set by the reg service. 
 Now that it's changed to `testsocialevent.WithTargetCluster(targetCluster))` the case where targetCluster is `""` should result in UserSignup with empty TargetCluster